mmdet/models/dense_heads/base_dense_head.py

```diff
@@ -76,3 +77,185 @@ def simple_test(self, feats, img_metas, rescale=False):
+                    # since mmdet v2.0
+                    # BG cat_id: num_class
+                    max_scores, _ = scores[..., :-1].max(-1)
+
+                _, topk_inds = max_scores.topk(nms_pre)
+                batch_inds = torch.arange(batch_size).view(
+                    -1, 1).expand_as(topk_inds)
+                # Avoid onnx2tensorrt issue in https://github.com/NVIDIA/TensorRT/issues/1134 # noqa: E501
+                # Mind k<=3480 in TensorRT for TopK
+                transformed_inds = scores.shape[1] * batch_inds + topk_inds
+                scores = scores.reshape(
+                    -1, self.num_classes)[transformed_inds].reshape(
+                        batch_size, -1, self.num_classes)
+                bbox_pred = bbox_pred.reshape(-1, 4)[transformed_inds].reshape(
+                    batch_size, -1, 4)
+                # prioris has shape (batch, num_prioris, 2) in FCOS, FOVEA etc. # noqa: E501
+                # prioris has shape (batch, num_prioris, 4) in Retina, ATSS etc. # noqa: E501
+                prioris = prioris.reshape(
+                    -1, prioris.size(-1))[transformed_inds].reshape(
+                        batch_size, -1, prioris.size(-1))
+                if with_score_factors:
+                    score_factor = score_factor.reshape(
+                        -1, 1)[transformed_inds].reshape(batch_size, -1, 1)
+
+            bboxes = self.bbox_coder.decode(
+                prioris, bbox_pred, max_shape=img_shapes)
+            mlvl_bboxes.append(bboxes)
+            mlvl_scores.append(scores)
+
+            if with_score_factors:
+                mlvl_cls_score_factors.append(score_factor)
+
+        batch_mlvl_bboxes = torch.cat(mlvl_bboxes, dim=1)
+        batch_mlvl_scores = torch.cat(mlvl_scores, dim=1)
+        if with_score_factors:
+            batch_score_factors = torch.cat(mlvl_cls_score_factors, dim=1)
+            batch_mlvl_scores = batch_mlvl_scores * batch_score_factors
+        # Replace multiclass_nms with ONNX::NonMaxSuppression in deployment
+        if with_nms:
+            from mmdet.core.export import add_dummy_nms_for_onnx
+            # ignore background class
+            if not self.use_sigmoid_cls:
+                num_classes = batch_mlvl_scores.shape[2] - 1
+                batch_mlvl_scores = batch_mlvl_scores[..., :num_classes]
+            max_output_boxes_per_class = cfg.nms.get(
+                'max_output_boxes_per_class', 200)
+            iou_threshold = cfg.nms.get('iou_threshold', 0.5)
+            score_threshold = cfg.score_thr
+            nms_pre = cfg.get('deploy_nms_pre', -1)
+            return add_dummy_nms_for_onnx(batch_mlvl_bboxes, batch_mlvl_scores,
+                                          max_output_boxes_per_class,
+                                          iou_threshold, score_threshold,
+                                          nms_pre, cfg.max_per_img)
+        else:
```

Comment:
Have you tested with ` with_nms=False` when exporting to ONNX?

Reply:
Not yet

Reply:
we may need to  test whether it could export and how the outputs are organized when `with_nms=False`